#! /bin/sh /usr/share/dpatch/dpatch-run
## 43_check_ntp_segfaults.dpatch by Thierry Carrez <thierry.carrez@ubuntu.com>
##
## DP: Fixes check_ntp and check_ntp_peer segfaults (LP: #291265)
## Patch originally from Thomas Guyot, extracted from upstream SVN at:
## http://nagiosplug.svn.sourceforge.net/viewvc/nagiosplug?view=rev&revision=2086

@DPATCH@
diff -urNad nagios-plugins-1.4.12~/plugins/check_ntp.c nagios-plugins-1.4.12/plugins/check_ntp.c
--- nagios-plugins-1.4.12~/plugins/check_ntp.c	2008-05-07 10:02:42.000000000 +0000
+++ nagios-plugins-1.4.12/plugins/check_ntp.c	2008-11-19 16:38:06.000000000 +0000
@@ -198,7 +198,7 @@
 /* NTP control message header is 12 bytes, plus any data in the data
  * field, plus null padding to the nearest 32-bit boundary per rfc.
  */
-#define SIZEOF_NTPCM(m) (12+ntohs(m.count)+((m.count)?4-(ntohs(m.count)%4):0))
+#define SIZEOF_NTPCM(m) (12+ntohs(m.count)+((ntohs(m.count)%4)?4-(ntohs(m.count)%4):0))
 
 /* finally, a little helper or two for debugging: */
 #define DBG(x) do{if(verbose>1){ x; }}while(0);
diff -urNad nagios-plugins-1.4.12~/plugins/check_ntp_peer.c nagios-plugins-1.4.12/plugins/check_ntp_peer.c
--- nagios-plugins-1.4.12~/plugins/check_ntp_peer.c	2008-05-07 10:02:42.000000000 +0000
+++ nagios-plugins-1.4.12/plugins/check_ntp_peer.c	2008-11-19 16:38:06.000000000 +0000
@@ -130,7 +130,7 @@
 /* NTP control message header is 12 bytes, plus any data in the data
  * field, plus null padding to the nearest 32-bit boundary per rfc.
  */
-#define SIZEOF_NTPCM(m) (12+ntohs(m.count)+((m.count)?4-(ntohs(m.count)%4):0))
+#define SIZEOF_NTPCM(m) (12+ntohs(m.count)+((ntohs(m.count)%4)?4-(ntohs(m.count)%4):0))
 
 /* finally, a little helper or two for debugging: */
 #define DBG(x) do{if(verbose>1){ x; }}while(0);
